### PR TITLE
feat: LambdaFunctionUrlAuth rule

### PR DESCRIPTION
**Pull request by dboyd13**
| Rule ID            | Cause                                                                                                                                                  | Explanation                                                                                                                                                                                                                                                                                                                               |
| ------------------ | ------------------------------------------------------------------------------------------------------------------------------------------------------ | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
| LambdaFunctionUrlAuth | The Lambda Function URL is open to the public as no authentication is implemented | AWS Lambda Function URLs allow you to invoke your function via a HTTPS end-point, setting the authentication to NONE allows anyone on the internet to invoke your function. | 

Additionally updates minimum cdk version to `2.21.0` for `CfnUrl` L1 construct

### CODE REVIEW

File: RULES.md [hunk 1]
```diff
@@ -696,6 +696,12 @@ Unimplemented rules from the AWS PCI DSS 3.2.1 Conformance Pack.
 | [ssm-document-not-public](https://docs.aws.amazon.com/config/latest/developerguide/ssm-document-not-public.html)                                                                 | SSM documents are not public.                                                                                                       | 1.2, 1.2.1, 1.3, 1.3.1, 1.3.2, 1.3.4, 1.3.6, 2.2.2                                                                |
 | [vpc-sg-open-only-to-authorized-ports](https://docs.aws.amazon.com/config/latest/developerguide/vpc-sg-open-only-to-authorized-ports.html)                                       | The VPC Security Group restricts IPv4 TCP traffic on unauthorized ports.<sup>[2](#vpc-sg-open-only-to-authorized-ports)</sup>       | 1.2, 1.2.1, 1.3, 1.3.1, 1.3.2, 2.2.2                                                                              |
 
+## Rules not in packs
```
Comment (dontirun) [inline RULES.md:699]:
```suggestion
## Additional Rules

A collection of community rules that are not currently included in any of the included packs.
```

Comment (dontirun) [reply RULES.md:699]: Adding some details here to clarify

Comment (dontirun) [inline RULES.md:699]: I think we should add a reference to the `README` about this. Perhaps under https://github.com/cdklabs/cdk-nag#available-packs section or a related section

File: RULES.md [hunk 2]
```diff
@@ -696,6 +696,12 @@ Unimplemented rules from the AWS PCI DSS 3.2.1 Conformance Pack.
...
+| ------------------ | ------------------------------------------------------------------------------------------------------------------------------------------------------ | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| LambdaFunctionUrlAuth | The Lambda Function URL is open to the public as no authentication is implemented | AWS Lambda Function URLs allow you to invoke your function via a HTTPS end-point, setting the authentication to NONE allows anyone on the internet to invoke your function. |
```
Comment (dontirun) [inline RULES.md:703]:
```suggestion
| LambdaFunctionUrlAuth | The Lambda Function URL allows for public, unauthenticated access.  | AWS Lambda Function URLs allow you to invoke your function via a HTTPS end-point, setting the authentication to NONE allows anyone on the internet to invoke your function. | 
```

Comment (dontirun) [reply RULES.md:703]: Taking [verbiage from the docs](https://docs.aws.amazon.com/lambda/latest/dg/urls-auth.html)

**Review by dontirun — CHANGES_REQUESTED** [review]
Looks good to me after some minor documentation changes and additions

File: RULES.md (hunk 1 above)
Comment (dontirun) [reply RULES.md]: @dbyod13 this needs to be addressed

Comment (dboyd13) [reply RULES.md]: Apologies, missed that one. Added proposed commit now

**Review by dontirun — APPROVED** [review]

**Pull request merged**